src/control/lib/control/auto.go

```diff
@@ -566,38 +571,39 @@ func getCPUDetails(log logging.Logger, numaSSDs numaSSDsMap, coresPerNuma int) (
 
 type newEngineCfgFn func(int) *engine.Config
 
-func defaultEngineCfg(idx int) *engine.Config {
+// DefaultEngineCfg is a standard baseline for the config generate to start from.
+func DefaultEngineCfg(idx int) *engine.Config {
 	return engine.NewConfig().
 		WithTargetCount(defaultTargetCount).
 		WithLogFile(fmt.Sprintf("%s.%d.log", defaultEngineLogFile, idx))
 }
 
-// genConfig generates server config file from details of network, storage and CPU hardware after
+// GenConfig generates server config file from details of network, storage and CPU hardware after
 // performing some basic sanity checks.
-func genConfig(log logging.Logger, newEngineCfg newEngineCfgFn, accessPoints []string, nd *networkDetails, sd *storageDetails, ccs numaCoreCountsMap) (*config.Server, error) {
-	if nd.engineCount == 0 {
+func GenConfig(log logging.Logger, newEngineCfg newEngineCfgFn, accessPoints []string, nd *NetworkDetails, sd *StorageDetails, ccs numaCoreCountsMap) (*config.Server, error) {
```

Comment:
Sort of confusing that both this and ConfigGenerate are exported functions in the same package. This makes me lean toward putting the shared functions in a separate package too.